Makefile

```diff
@@ -0,0 +1,78 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect C++ sources, and link if necessary
+CPPSRC := $(wildcard src/*.cc)
+
+ifeq (, $(CPPSRC))
+	ADDITIONALLIBS := 
+else
+	ADDITIONALLIBS := -lc++
+endif
+
+# collect sources
+SRC := $(wildcard src/*.c)
+SRC += $(CPPSRC)
+OBJ := $(addsuffix .o,$(basename $(SRC)))
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+# define default flags, and override to append mandatory flags
+# CPPFLAGS apply to both c and c++
+# CFLAGS only affects c
+CPPFLAGS ?= -O3 -Wall -Wextra -Werror
+override CFLAGS += -std=gnu99
+override CPPFLAGS += -fPIC -Ilib/src -Ilib/include
```

Comment:
I don't think we need these two `-I` flags, because there won't be a `lib` folder for the generated parsers.

I would also 🔥 the comments about how CPPFLAGS differs from CFLAGS, since I think that is a standard Make-ism.

Reply:
Thanks so much for looking so closely!